CMakeLists.txt

```diff
@@ -947,8 +947,23 @@ if(CONFIG_USERSPACE)
   set(PROCESS_GPERF ${ZEPHYR_BASE}/scripts/build/process_gperf.py)
 endif()
 
+# Backwards compatible: allow manual selection of C standard using a global property
```

Comment:
I think we could deprecated the use of `CSTD` and print a deprecation warning when defined.